api/v1alpha1/packagebundle_webhook.go

```diff
@@ -50,10 +55,19 @@ func (r *PackageBundle) ValidateDelete() error {
 }
 
 func (r *PackageBundle) validate() error {
-	valid, digest, yml, err := signature.ValidateSignature(r, signature.EksaDomain)
-	if err == nil && !valid {
+	//TODO Turn this into fields in a new CRD used for signature validation configuration
+	keyOverride := os.Getenv(PublicKeyEnvVar)
```

Comment:
This might be overkill, but in the name of visibility and reproducibility... What if we inserted this value at build time?

Makefile (pseudo):
```
build:
    go build -ldflags "-X api.v1alpha1.keyOverride=$(data here, from env or catting a file)" .
```
The advantage of this version, is that the binary contains the public key, it could be identified with `strings` or the like, and there's no doubt about which key was used. With the env var approach, the key could change with each run of the code. We could have the dev/testing key in the code itself, and override with prod/staging at build time.

I guess another alternative is to, at build time, generate a file that defines the public key as a constant. That's in some ways even better, since as a constant, it can't be changed. Though a non-exported var also can't be changed (by code outside of the package that we control at least)...

Anyway, I'm not asking for a change here, more just brainstorming possible improvements.

Reply:
Ultimately we want to allow users to define trusted keys and domains so all of this would make its way to a CRD. If we keep a hard-coded copy of the key, we could use what you propose to override it in different build stages. The downside is however that we'd end up with non-matching digests across environments. We would not be able to simply promote a build.